lambdas/checklist/evaluate.py

```diff
@@ -106,14 +106,17 @@ def retrieve_candidate_github_data(github_client: GithubClient) -> list:
     :param github_client: The GitHub client.
     :return: The candidate's GitHub repository data.
     """
-    pinned_repositories = github_client.get_pinned_repositories_name()
+    n_repositories = 8
```

Comment:
What happens if we go over 8? Will the performance be affected or we stop at 8 because its sufficient?

Reply:
As long as we use GPT-3.5, reading eight repositories is kind of the maximum. (depending on the sizes, but having 10 or more small repos shouldn't improve the score)

Reply:
https://platform.openai.com/docs/models/models
https://openai.com/pricing

we can use GPT-4 turbo or other APIs to increase token

Reply:
Gemini Pro: max 30K, ≈ $0.001 / 1K token
GPT-3.5 Turbo: max 16K, $0.001 / 1K token
GPT-4 Turbo: max 128K, $0.001 / 1K token
Anthropic: max 100K, $0.02? / 1K token


one evaluation for 8 repos (≈max): 16K

https://ai.google.dev/pricing
https://www-files.anthropic.com/production/images/model_pricing_dec2023.pdf

Reply:
Ok I think we're good with GPT3.5 Turbo for now